tests/test_click_cli.py

```diff
@@ -167,3 +167,115 @@ def test_cli_cleanup_command_line_has_higher_priority(tmp_path):
     with context_for_test(app_name):
         run_and_check(cli_args, app_name, build_dir)
     assert not build_dir.exists()
+
+
+def test_cli_module_name_conflict(tmp_path):
+    app_name = 'app_simple'
+    build_dir = tmp_path / 'build'
+    config_file = tmp_path / 'config.yaml'
+    config_content = {'app_module': 'app'}
+    with open(config_file, mode='w') as file:
+        safe_dump(config_content, stream=file)
+    cli_args = ['app', str(build_dir), '--config', config_file]
+
+    with pytest.raises(click.UsageError):
+        run_and_check(cli_args, app_name, build_dir)
+
+
+def test_cli_dest_path_conflict(tmp_path):
+    app_name = 'app_simple'
+    build_dir = tmp_path / 'build'
+    config_file = tmp_path / 'config.yaml'
+    config_content = {'output': {'type': 'dir', 'dir': str(build_dir)}}
+    with open(config_file, mode='w') as file:
+        safe_dump(config_content, stream=file)
+    cli_args = ['app', str(build_dir), '--config', config_file]
+
+    with pytest.raises(click.UsageError):
+        run_and_check(cli_args, app_name, build_dir)
+
+
+def test_cli_module_name_as_argument_no_dest_path(tmp_path):
+    app_name = 'app_simple'
+    build_dir = tmp_path / 'build'
+    cli_args = ['app']
+
+    with pytest.raises(click.UsageError):
+        run_and_check(cli_args, app_name, build_dir)
+
+
+def test_cli_dest_path_as_argument_no_module_name(tmp_path):
+    app_name = 'cli_module_name'
+    build_dir = tmp_path / 'build'
+    cli_args = [str(build_dir)]
+
+    with pytest.raises(ModuleNotFoundError):
```

Comment:
This error occurs because of bad user usage, Do we want to test this case?

Reply:
Yes. In such cases we want to tell the user what's wrong.
(For freezeyt, `ModuleNotFoundError` is good. If this was aimed at non-programmers, we'd show a more human-friendly message and test that.)